packages/react-mutation-mapper/src/component/track/PtmTrack.tsx

```diff
@@ -99,18 +134,33 @@ export default class PtmTrack extends React.Component<PtmTrackProps, {}> {
             .ptmDataByProteinPosStart.result;
 
         if (ptmDataByProteinPosStart && !_.isEmpty(ptmDataByProteinPosStart)) {
-            return _.keys(ptmDataByProteinPosStart)
-                .filter(position => Number(position) >= 0)
-                .map(position => ({
-                    codon: Number(position),
-                    color: ptmColor(ptmDataByProteinPosStart[Number(position)]),
-                    tooltip: (
-                        <PtmTooltip
-                            ptms={ptmDataByProteinPosStart[Number(position)]}
-                            pubMedCache={this.props.pubMedCache}
-                        />
-                    ),
-                }));
+            return _.compact(
```

Comment:
this whole thing could be written as a single reduce function.  which would save 3 iteration cycles